### PR TITLE
Added ArmorStand and EntityEquipment mocks

### PR DESCRIPTION
**Pull request by TheBusyBiscuit**
# Description
This pull request adds an ArmorStand mock and an EntityEquipment mock.

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Version number updated in `settings.gradle`.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
Before the pull request can be accepted, the version number stored in settings.gradle should be updated.
MockBukkit uses semantic-versioning (https://semver.org/)

 - If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
 - If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)
 - Make sure that unit tests are added which test the relevant changes.
 - Make sure that the changes follow the existing code format.

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

The version number can be found in `settings.gradle`.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/LivingEntityMock.java [hunk 1]
```diff
@@ -476,13 +475,6 @@ public void setRemoveWhenFarAway(boolean remove)
 		throw new UnimplementedOperationException();
 	}
 
-	@Override
-	public EntityEquipment getEquipment()
-	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
-	}
-
```
Comment (seeseemelk) [inline src/main/java/be/seeseemelk/mockbukkit/entity/LivingEntityMock.java:485]: Why did you move this to `SimpleMobMock` and `SimpleMonsterMock`?

**Review by seeseemelk — APPROVED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/entity/LivingEntityMock.java (hunk 1 above)
Comment (TheBusyBiscuit) [reply src/main/java/be/seeseemelk/mockbukkit/entity/LivingEntityMock.java:485]: Because `EntityEquipment` is now mocked too, `#getEquipment()`  should not default to an `UnimplementedOperationException` in my opinion.
That's why I made the method abstract, so it can be decided on a per-class basis.
`SimpleMobMock` and `SimpleMonsterMock` currently do not implement the equipment, so they will now default to `UnimplementedOperationException` as before.

**Review by Vinetos — APPROVED** [review]

**Comment by Vinetos** [comment]
Hello ! What is missing ? 
I appreciate this feature !

**Comment by TheBusyBiscuit** [comment]
@Vinetos Glad to see you like it.
I could merge it already, though I'd love to get a final look and okay from seeseemelk first since he had a question about the changes and has yet to respond to my answer.
He's currently busy with other things, so it may take a few days for him to respond but don't worry, my changes will be merged as soon as possible.
